### PR TITLE
Hit [POST]/[DELETE] /posts/:post_id/likes endpoint

### PR DESCRIPTION
**Pull request by ntan03**
Current user liked top two posts:
image

Other user only liked first post:
image

### CODE REVIEW

**Review by ZohaibManzoor00 — APPROVED** [review]
For the future break this into two tasks. Post like, Delete like. 
Other than that, looks good!

**Review by Matt-Tan15 — DISMISSED** [review]

File: src/shared/components/post/PostFooter.tsx [hunk 1]
```diff
@@ -3,19 +3,40 @@ import { Box, IconButton, Typography } from "@mui/material";
...
+
+function PostFooter({ id, likes, comments }: BasePostFooterType) {
   const {
     themeColor: { color, navButtons },
   } = useTheme();
+  const user = useSelector(getCurrentUser);
   const [showComment, setShowComment] = useState(false);
   const [displayLikeCount, setDisplayLikeCount] = useState(likes.length);
-  const [isLiked, setIsLiked] = useState(false);
+  const [isLiked, setIsLiked] = useState(
```
Comment (raylu511) [inline src/shared/components/post/PostFooter.tsx:19]: unneccessary state here. liked is a prop so this comment will rerender with the new likes array.
you can just do
const isLiked = likes.find(like => like.id === user.id)

File: src/shared/components/post/PostFooter.tsx [hunk 2]
```diff
@@ -3,19 +3,40 @@ import { Box, IconButton, Typography } from "@mui/material";
...
+    likes.find((like) => like.id === user.id) !== undefined
+  );
+
+  const handleLike = async () => {
+    const endpoint = process.env.REACT_APP_BASE_URL + `/posts/${id}/likes`;
+    await axios.post(endpoint);
+
+    setDisplayLikeCount(displayLikeCount + 1);
```
Comment (raylu511) [inline src/shared/components/post/PostFooter.tsx:27]: this state is unneccessary here too. look at comment above

File: src/shared/components/post/PostFooter.tsx (hunk 1 above)
Comment (ntan03) [reply src/shared/components/post/PostFooter.tsx:19]: Ah, I see. I currently only have the PostFooter handling all the likes/dislikes, but I do see how the props can be used to rerender the like button.

File: src/features/feed/components/Feed.tsx [hunk 3]
```diff
@@ -22,6 +22,15 @@ function Feed() {
   const {
     themeColor: { color, navButtons, backgroundColor },
   } = useTheme();
+
+  const handleLikeDislike = (id: any, userId: any, isLike: any) => {
+    var postLikes = [...feed[feed.findIndex((p) => p.id === id)].likes];
```
Comment (ZohaibManzoor00) [inline src/features/feed/components/Feed.tsx:27]: let instead of var

**Review by ZohaibManzoor00 — APPROVED** [review]
Looks good

File: src/shared/components/post/Post.tsx [hunk 4]
```diff
@@ -40,6 +41,12 @@ function Post({ basePostProps, onDelete }: PostProps) {
     timestamp,
   };
 
+  const postFooterProps = {
+    id,
+    likes,
+    comments,
+  };
+
   // The idea is that the Post component will handle the state of its comments
```
Comment (Matt-Tan15) [inline src/shared/components/post/Post.tsx:50]: nit: remove comment

**Review by Matt-Tan15 — APPROVED** [review]

**Comment by raylu511** [comment]
If this PR has changes that's unrelated to what you did, you should create a new branch and push that up

**Pull request closed without merging**